docs/developer-docs/smart-contracts/advanced-features/http-gateways.mdx

```diff
@@ -0,0 +1,105 @@
+---
+keywords: [advanced, tutorial, http, http gateways, gateways]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# HTTP gateways
+
+<MarkdownChipRow labels={["Advanced", "Tutorial"]} />
+
+## Overview
+
+The ICP HTTP gateway protocol enables conventional HTTP clients to interact with the ICP network. Through the HTTP gateway, software such as web browsers can retrieve and digest canister code and assets, such as HTML, JavaScript, images, or videos. The HTTP gateway makes this workflow possible by translating standard HTTP requests into API canister calls that ICP can understand.
+
+There are several HTTP gateway nodes running on ICP, but it should be noted that an HTTP gateway node could become compromised, creating a threat for canisters. Compromised gateways are threats even for canisters that are considered 'secure', such that they use certify assets, use a default client, and do not maintain their own certification checking.
+
+For a more secure solution, it is possible to run your own HTTP gateway instance locally.
+
+### Terminology
+
+- **Request**: A message sent to a server or endpoint.
+
+- **Client**: A service that is able to send an HTTP formatted request and receive a response from a server.
```

Comment:
We seem to say "HTTP client" elsewhere in the text.

Reply:
Each term defined here is referenced with `HTTP` throughout the doc. We could include `HTTP` in each definition, but it may feel redundant.